src/kernels/debugger/kernelDebugAdapterBase.ts

```diff
@@ -189,10 +200,14 @@ export abstract class KernelDebugAdapterBase implements DebugAdapter, IKernelDeb
     }
 
     public async disconnect() {
-        await this.session.customRequest('disconnect', { restart: false });
-        this.endSession.fire(this.session);
-        this.disconnected = true;
-        this.kernel?.removeEventHook(this.kernelEventHook);
+        if (!this.disconnected) {
+            if (this.debugService.activeDebugSession === this.session) {
+                await this.session.customRequest('disconnect', { restart: false });
+            }
+            this.endSession.fire(this.session);
+            this.disconnected = true;
```

Comment:
I think we should move this property initialization to the very top (just after checking it), else this method `disconnect` can be called multiple times and we could end up sending multiple `disconnect` requests

Reply:
Ah good point.